src/pages/workspaces/workspace/Dashboard/Dashboard.ts

```diff
@@ -86,25 +92,19 @@ const DashboardAuthContainer = (props: WorkspaceDashboardProps): ReactNode => {
     ],
     [signInStatus === 'signedOut' && isFeaturedWorkspace(), () => h(DashboardPublic, props)],
     [signInStatus === 'signedOut', () => h(SignIn)],
-    () => h(WorkspaceDashboard, props)
+    () => h(WorkspaceDashboardPage, props)
   );
 };
 
 interface WorkspaceDashboardProps {
   namespace: string;
   name: string;
-}
-
-interface WorkspaceDashboardComponentProps extends WorkspaceDashboardProps {
   refreshWorkspace: () => void;
   storageDetails: StorageDetails;
   workspace: Workspace;
 }
 
-const WorkspaceDashboardForwardRefRenderFunction = (
-  props: WorkspaceDashboardComponentProps,
-  ref: ForwardedRef<unknown>
-): ReactNode => {
+const WorkspaceDashboard = forwardRef((props: WorkspaceDashboardProps, ref: ForwardedRef<unknown>): ReactNode => {
```

Comment:
I'm still not clear what the `forwardRef` does for us (why we need it at all). Can you explain?

Reply:
I guess it is related to forcing a refresh? But who is forcing refreshes of the Dashboard?

Reply:
`forwardRef` lets WorkspaceDashboard customize the handle it exposes via a ref using `useImperativeHandle`.

https://react.dev/reference/react/useImperativeHandle

Components wrapped by `wrapWorkspace` are expected to have a `refresh` method on their ref handle, which WorkspaceContainer calls when the current tab is clicked.

Reply:
I'm confused that we have 3 names for what I think always has to be the same thing-- `WorkspaceWrapperProps`, `WorkspaceDashboardProps`, and `DashboardAuthContainerProps`

Reply:
The workspace wrapper component and DashboardAuthContainer do take the same props: workspace namespace and name.

WorkspaceDashboard takes those and the additional workspace info loaded by WorkspaceContainer.

I think a little duplication is better than a dependency, so I defined DashboardAuthContainerProps as its own interface, though it is the same as WorkspaceWrapperProps. I think that makes the code easier to read since the props type appears right above the component. Previously, things were defined all out of order so it was very hard to follow. The components are still out of order (DashboardAuthContainer depends on WorkspaceDashboardPage).

I'm also imagining that a natural next step is to split these components out into their own modules.

Reply:
What if we declare an interface just for namespace/name, and have the other ones extend it? What I'm getting at is that I believe all these components need `namespace` and `name`, and changing one of them to not have one or both of those would break things. But right now it seems like it also could just be coincidental that they all  have the same properties.

Reply:
I think that would be ok here, but if we split the components into separate modules (as I think we're likely to do), we'd end up either back to duplicating types like this PR or importing types, which I think makes the code more difficult to understand.